### PR TITLE
Fix expectation calculation for Poisson Distribution

### LINKED ISSUE #27344: Why doesn't expectation of Poisson random variable just return lambda?
I find the behavior of some aspects of Sympy to be utterly bizarre.  For example, take these two lines:

``` python
>>> E(Poisson('xxx', np.float64(4.72544290380919e-11)))
4.72544290358589e-11*exp(472544290380919/10000000000000000000000000)
>>> E(Poisson('xxx', np.float64(4.725442903809197e-11)))
...
ValueError: Could not calculate 2 terms for 0.999999999952746*exp(4.7254429038092e-11*exp(_x))
```

Clearly some kind of floating point error is occurring here, but why?

The expectation of a Poisson random variable is λ, so why not just return the original argument?  Instead `E(...)` is returning `e^λ * (λ / e^λ)` and creating precision issues.

Does it have something to do with the generation of Nth moment, and if so why isn't it reduced to λ before evaluation?

### PR DESCRIPTION
**Pull request by RushabhMehta2005**
#### References to other Issues or PRs

Fixes #27344

#### Brief description of what is fixed or changed
This PR overrides the expectation method of SingleDiscreteDistribution in PoissonDistribution, which simply returns the correct expected value. 
A test has been added corresponding to this change.
#### Release Notes

* stats
  * Fixed expectation calculation for Poisson Distribution

### CODE REVIEW

**Comment by RushabhMehta2005** [comment]
.mailmap check is failing for some other contributor.

**Comment by jherkenhoff** [comment]
> .mailmap check is failing for some other contributor.

Same for me in my PR opened a few hours ago (#27346)

**Comment by RushabhMehta2005** [comment]
Please review this, thank you.

File: sympy/stats/drv_types.py [hunk 1]
```diff
@@ -540,6 +540,13 @@ def _characteristic_function(self, t):
     def _moment_generating_function(self, t):
         return exp(self.lamda * (exp(t) - 1))
 
+    def expectation(self, expr, var, evaluate=True, **kwargs):
+        if evaluate:
+            if expr == var:
+                return self.lamda
+            if isinstance(expr, FallingFactorial) and expr.args[1].is_integer and expr.args[1].is_positive and expr.args[0] == var:
```
Comment (haru-44) [inline sympy/stats/drv_types.py:547]: Please break the lines appropriately, as the current lines are too long.  
Also, please add tests related to this code.

Comment (RushabhMehta2005) [reply sympy/stats/drv_types.py:547]: I have added a test for the E(X) = lambda case. I will add for the rest.

File: sympy/stats/tests/test_discrete_rv.py [hunk 2]
```diff
@@ -50,6 +51,17 @@ def test_Poisson():
     assert isinstance(E(2*x, evaluate=False), Expectation)
     # issue 8248
     assert x.pspace.compute_expectation(1) == 1
+    try:
+        import numpy as np
+    except ImportError:
+        skip("numpy not installed")
+    y = Poisson('y', np.float64(4.72544290380919e-11))
```
Comment (haru-44) [inline sympy/stats/tests/test_discrete_rv.py:58]: I'm sorry. I should have pointed it out yesterday, but please make sure to leave the issue number as a comment.

Comment (RushabhMehta2005) [reply sympy/stats/tests/test_discrete_rv.py:58]: I will do that. Could you briefly explain when do newly added tests need an issue number accompanying them?

**Comment by oscarbenjamin** [comment]
The slow test timeout is something that happens periodically.

The other test failure is with the mpmath master branch and must be because of some new change in mpmath.

**Comment by RushabhMehta2005** [comment]
The test fail issue is persistent.

**Comment by oscarbenjamin** [comment]
The other mpmath problem got fixed. I think that what remains is different:
```
    def test_subs():
>       assert NS('besseli(-x, y) - besseli(x, y)', subs={x: 3.5, y: 20.0}) == \
            '-4.92535585957223e-10'
E       AssertionError: assert '0.e-116' == '-4.92535585957223e-10'
E         
E         - -4.92535585957223e-10
E         + 0.e-116
```
I think actually this test is incorrect and is asserting something that only happens because of inaccuracy in mpmath that is now improved in mpmath master.

**Comment by RushabhMehta2005** [comment]
Should this test be removed?

File: sympy/stats/tests/test_discrete_rv.py (hunk 2 above)
Comment (haru-44) [reply sympy/stats/tests/test_discrete_rv.py:59]: As with any code, making it easy to read is essential. This means reducing cognitive load. For instance, testing that the expected value of $Poisson(\lambda)$ is $\lambda$ seems natural (so I don’t think leaving a comment is necessary). On the other hand, someone reading a test case involving an unfamiliar value like `4.72544290380919e-11` might wonder, "Why this number?" While there are no strict rules, I believe comments are necessary to ensure that someone unfamiliar with the context won't get stuck while reading the code.

**Comment by haru-44** [comment]
looks good. thanks.
I will merge it if no one else points it out.

> Should this test be removed?

It is possible to create a separate PR, but at the very least, this is not an issue to be fixed here.

**Comment by oscarbenjamin** [comment]
The mpmath issues have been fixed in mpmath:
https://github.com/mpmath/mpmath/pull/889#discussion_r1873667039

File: sympy/stats/tests/test_discrete_rv.py (hunk 2 above)
Comment (RushabhMehta2005) [reply sympy/stats/tests/test_discrete_rv.py:59]: That makes complete sense! Thank you for the explaination, I will keep this in mind with my further contributions.

**Comment by RushabhMehta2005** [comment]
> looks good. thanks. I will merge it if no one else points it out.

Thank you!

> It is possible to create a separate PR, but at the very least, this is not an issue to be fixed here.

Alright, different PRs for seperate tasks.

File: sympy/stats/tests/test_discrete_rv.py (hunk 2 above)
Comment (Evidlo) [reply sympy/stats/tests/test_discrete_rv.py:59]: Should this be `4.725442903809197e-11` ? (ending in 7)

`4.72544290380919e-11` was the case that worked even before this patch.

Comment (RushabhMehta2005) [reply sympy/stats/tests/test_discrete_rv.py:59]: I assume the exact numbering is not the issue here, more about the handling of E(x) queries.

Comment (haru-44) [reply sympy/stats/tests/test_discrete_rv.py:59]: I overlooked it. I'll fix it.

Comment (RushabhMehta2005) [reply sympy/stats/tests/test_discrete_rv.py:59]: Even I missed that, sorry.

**Pull request merged**